src/utils/useAuthentication/index.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+import { logout, startTTH, TStartTTH } from '../TwoTokenHandshake';
+
+export const useAuthentication = (authentication: Partial<TStartTTH>) => {
+  const { userId, performBEDHandshakeCallback } = authentication;
```

Comment:
Can we maybe change `performBEDHandshakeCallback` to something more declarative? Like `onUserChange` or `onAuthChange`? Not super important though...